apps/server/src/modules/tool/tool-launch/service/tool-launch.service.ts

```diff
@@ -84,22 +87,25 @@ export class ToolLaunchService {
 		};
 	}
 
-	private isToolStatusLaunchableOrThrow(
+	private async isToolStatusLaunchableOrThrow(
 		userId: EntityId,
 		externalTool: ExternalTool,
 		schoolExternalTool: SchoolExternalTool,
 		contextExternalTool: ContextExternalToolLaunchable
-	): void {
-		const status: ContextExternalToolConfigurationStatus = this.toolVersionService.determineToolConfigurationStatus(
-			externalTool,
-			schoolExternalTool,
-			contextExternalTool
-		);
+	): Promise<void> {
+		const status: ContextExternalToolConfigurationStatus =
+			await this.toolConfigurationStatusService.determineToolConfigurationStatus(
+				externalTool,
+				schoolExternalTool,
+				contextExternalTool,
+				userId
+			);
 
 		if (
 			status.isOutdatedOnScopeSchool ||
 			status.isOutdatedOnScopeContext ||
 			status.isDeactivated ||
+			status.isNotLicensed ||
```

Comment:
This is also checked in the tool-launch.uc.ts line 47, 82 and can be removed there.

Reply:
you're right-> removed! But this could make the code harder to maintain (easier to forget the notLicense state when the uc changes etc...)

Reply:
> can be removed there

not here

Reply:
😳 ...